### PR TITLE
fix: username in user

### PR DESCRIPTION
**Pull request by adlerhurst**
for a better overview I created a separate branch for user name.

### CODE REVIEW

File: internal/api/grpc/auth/user_converter.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func userViewFromModel(user *usr_model.UserView) *auth.UserView {
 		ResourceOwner:      user.ResourceOwner,
 		LoginNames:         user.LoginNames,
 		PreferredLoginName: user.PreferredLoginName,
+		// UserName: user.UserName TODO: remove comment
```
Comment (livio-a) [inline internal/api/grpc/auth/user_converter.go:60]: ?

File: internal/user/model/user_human.go [hunk 2]
```diff
@@ -41,18 +38,18 @@ const (
...
+// 		return caos_errors.ThrowPreconditionFailed(nil, "MODEL-zSH7j", "Errors.Users.OrgIamPolicyNil")
+// 	}
+// 	if policy.UserLoginMustBeDomain && strings.Contains(u.UserName, "@") {
+// 		return caos_errors.ThrowPreconditionFailed(nil, "MODEL-se4sJ", "Errors.User.EmailAsUsernameNotAllowed")
+// 	}
+// 	if !policy.UserLoginMustBeDomain && u.Profile != nil && u.UserName == "" && u.Email != nil {
+// 		u.UserName = u.EmailAddress
+// 	}
+// 	return nil
+// }
```
Comment (livio-a) [inline internal/user/model/user_human.go:52]: remove it as you moved it to user.go

File: internal/api/grpc/auth/user_converter.go (hunk 1 above)
Comment (fgerschwiler) [inline internal/api/grpc/auth/user_converter.go:60]: remove todo

**Review by livio-a — COMMENTED** [review]
there are two code snipped commented on your side... but good otherwise

and you deleted the specific `creates` which would then differ from creating applications

File: internal/user/model/user.go [hunk 3]
```diff
@@ -56,11 +61,24 @@ func (u *User) IsLocked() bool {
 }
 
 func (u *User) IsValid() bool {
-	if u.Human == nil && u.Machine == nil {
+	if u.Human == nil && u.Machine == nil || u.UserName == "" {
```
Comment (fgerschwiler) [inline internal/user/model/user.go:64]: check username for " "

File: internal/user/model/user_human.go (hunk 2 above)
Comment (fgerschwiler) [inline internal/user/model/user_human.go:52]: remove comment

File: pkg/grpc/management/proto/management.proto [hunk 4]
```diff
@@ -136,25 +136,8 @@ service ManagementService {
...
-      permission: "user.write"
-    };
-  }
-
-  rpc CreateMachine(CreateMachineRequest) returns (UserResponse) {
     option (google.api.http) = {
-      post: "/users/machine"
+      post: "/users"
       body: "*"
     };
```
Comment (fgerschwiler) [inline pkg/grpc/management/proto/management.proto:142]: if we remove this, we have it in another way like in applications.

Comment (adlerhurst) [reply pkg/grpc/management/proto/management.proto:142]: it's the easiest way to handle username in subobjects

File: internal/user/model/user_human.go (hunk 2 above)
Comment (adlerhurst) [reply internal/user/model/user_human.go:52]: removed

Comment (adlerhurst) [reply internal/user/model/user_human.go:52]: removed

File: internal/api/grpc/auth/user_converter.go (hunk 1 above)
Comment (adlerhurst) [reply internal/api/grpc/auth/user_converter.go]: fixed

Comment (adlerhurst) [reply internal/api/grpc/auth/user_converter.go]: fixed

File: internal/user/model/user.go (hunk 3 above)
Comment (adlerhurst) [reply internal/user/model/user.go:64]: added in validation

File: internal/api/grpc/auth/user_machine_converter.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+	expirationDate, err := ptypes.TimestampProto(key.CreationDate)
+	logging.Log("MANAG-gluk7").OnError(err).Debug("unable to parse timestamp")
+
+	return &auth.MachineKeyView{
+		Id:             key.ID,
+		CreationDate:   creationDate,
+		ExpirationDate: expirationDate,
+		Sequence:       key.Sequence,
+		Type:           auth.MachineKeyType_MACHINEKEY_UNSPECIFIED, //TODO: type mapping
```
Comment (fgerschwiler) [inline internal/api/grpc/auth/user_machine_converter.go:41]: do type mapping

Comment (adlerhurst) [reply internal/api/grpc/auth/user_machine_converter.go:41]: I will implement it if i implement the keys. It doesnt make sense to return something not existing

File: pkg/grpc/management/proto/management.proto [hunk 6]
```diff
@@ -1356,7 +1356,7 @@ message Email {
 }
 
 message UniqueUserRequest {
-  string user_name = 1 [(validate.rules).string = {min_len: 1, max_len: 200}];
+  string user_name = 1 [(validate.rules).string.pattern = "^[[:graph:]]{1,200}$"];
```
Comment (livio-a) [inline pkg/grpc/management/proto/management.proto:1359]: this regex doesn't allow `é` or `ä`

shouldn't we just check for spaces at the beginning / end and trim them or return an error
IMHO if you insert a space, emoji or any other (byte) value which doesn't break our system, i'm fine with it

**Review by livio-a — CHANGES_REQUESTED** [review]

File: pkg/grpc/management/proto/management.proto (hunk 6 above)
Comment (adlerhurst) [reply pkg/grpc/management/proto/management.proto:1359]: changed regex. now all characters should be allowd (adföäüèê😀sdf) is a valid name

**Review by livio-a — APPROVED** [review]

**Pull request merged**